nemoguardrails/embeddings/embedding_providers/cohere.py

```diff
@@ -0,0 +1,104 @@
+import logging
+from typing import Any, List
+import os
+import cohere
+from tenacity import retry, stop_after_attempt, wait_fixed, before_sleep_log
+
+from . import EmbeddingModel
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class CohereEmbeddingModel(EmbeddingModel):
+    """Embedding model using Cohere API.
+
+    Args:
+        embedding_model (str): The name of the embedding model.
+
+    Attributes:
+        model (str): The name of the embedding model.
+        embedding_size (int): The size of the embeddings.
+
+    Methods:
+        encode: Encode a list of documents into embeddings.
+    """
+
+    def __init__(
+            self,
+            embedding_model: str,
+    ):
+        self.model = embedding_model
+        self.truncate = "NONE"
+
+        self.co_api_key = os.environ.get("COHERE_API_KEY")
+        if not self.co_api_key:
+            raise Exception("`COHERE_API_KEY` not provided")
+
+        self.client = None # shared client causes event loop closed error
```

Comment:
Not sure why this happens, in langchain they use a single client for Cohere embeddings:
https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/embeddings/cohere.py

Reply:
I rechecked, but I could not figure it out. I believe it is acceptable to have it created each time, considering it is a lightweight component.